### PR TITLE
feat(links)!: differentiate external from internal links

### LINKED ISSUE #197: Cookie Links to route to page (using NuxtLink)
### Describe the feature

At present we can have links for cookies where more information can be provided. Whether you configure this with a page route e.g. /MyCookiePolicy or an external route e.g. http://mysite/cookiepolicy the link is always rendered using an <a> tag, which means Nuxt will perform a full page load.
If would be nice if we could use Nuxt/Vue routing to close the CookieDialog and perform a Page navigation.

The fix could be something like this in the modal dialog:
            <NuxtLink
              v-if="!entry[0].startsWith('http')"
              :to="entry[0]"
              @click="isModalActive = false"
              >{{ entry[1] || entry[0] }}</NuxtLink
            >
            <a v-else :href="entry[0]">{{ entry[1] || entry[0] }}</a>

i.e. if its not an external link use Nuxt/Vue routing otherwise use the <a> tag and treat the link as external.

### Additional information

- [X] Would you be willing to help implement this feature?

### Final checks

- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

### PR DESCRIPTION
**Pull request by mouldt**
### 📚 Description

adds functionality to specify a cookie link to a local page inside your nuxt app or to an external full link.

"Resolves #197"

### 📝 Checklist

- [X] All commits follow the Conventional Commit format
- [X] The PR's title follows the Conventional Commit format

### CODE REVIEW

File: src/runtime/components/CookieControl.vue [hunk 1]
```diff
@@ -141,7 +141,13 @@
...
                               >
                                 <br />
-                                <a :href="entry[0]">{{
+                                <NuxtLink
+                                  v-if="!entry[0].startsWith('http')"
+                                  :to="entry[0]"
+                                  @click="isModalActive = false"
+                                  >{{ entry[1] || entry[0] }}</NuxtLink
+                                >
+                                <a v-else :href="entry[0]">{{
```
Comment (dargmuesli) [inline src/runtime/components/CookieControl.vue:150]: I think this `else` can be replaced by [NuxtLink's](https://nuxt.com/docs/api/components/nuxt-link#nuxtlink) `external` property.

Comment (mouldt) [reply src/runtime/components/CookieControl.vue]: updated the NuxtLink to use external

Comment (dargmuesli) [reply src/runtime/components/CookieControl.vue]: Please let me mark my comments as resolved once I confirmed the new implementation matches what I was thinking of :pray:

Comment (mouldt) [reply src/runtime/components/CookieControl.vue]: oops - apologies

File: src/runtime/components/CookieControl.vue [hunk 2]
```diff
@@ -141,9 +141,18 @@
                                 :key="entry[0]"
                               >
                                 <br />
-                                <a :href="entry[0]">{{
-                                  entry[1] || entry[0]
-                                }}</a>
+                                <NuxtLink
+                                  :to="entry[0]"
+                                  :external="
```
Comment (dargmuesli) [inline src/runtime/components/CookieControl.vue]: Actually, `external` is detected automatically and can be left out, I learned.

**Review by dargmuesli — COMMENTED** [review]
I've made some changes. Thank you very much for this suggestion! I'll merge it into `beta` because it might be a breaking change for some.

**Pull request merged**